### PR TITLE
Theora transport plugin

### PR DESCRIPTION
**Pull request by ivanpauno**
Added theora_image_transport and dependencies:
- ogg
- vorbis
- theora

### CODE REVIEW

**Review by meyerj — COMMENTED** [review]
Depends on `autoconf` and `automake` being installed, which already has been added to the docker container in https://github.com/Intermodalics/ros_android/pull/39/files#diff-ebacf6f6ae4ee68078bb16454b23247d. After installing `autoconf` manually, builds successfully.

I did not test the theora transport to verify whether it actually works. But the exported CMake configuration is lacking the transitive link dependencies to `libtheora.a` and `libogg.a`, so I would expect that a patch for [theora_image_transport/CMakeLists.txt](https://github.com/ros-perception/image_transport_plugins/blob/1.9.5/theora_image_transport/CMakeLists.txt) is needed to actually export `OpenCV`, `PC_THEORA` `PC_OGG`, `PC_THEORAENC` and `PC_THEORADEC` in the `catkin_package()` call.

**Comment by ivanpauno** [comment]
Thanks @meyerj.
I have added autoconf and automake to the Dockerfile. Also added required dependencies in a patch.
Now it is ready to merge.

File: get_library.sh [hunk 1]
```diff
@@ -63,6 +66,9 @@ elif [ $1 == 'poco' ]; then
 elif [ $1 == 'qhull' ]; then
     URL=http://www.qhull.org/download/qhull-2015-src-7.2.0.tgz
     COMP='gz'
+elif [ $1 == 'theora' ]; then
+    URL=http://downloads.xiph.org/releases/theora/libtheora-1.1.1.tar.bz2
+    COMP='bz2'
```
Comment (meyerj) [inline get_library.sh]: nit: remove trailing spaces

**Comment by meyerj** [comment]
Looks better:
```cmake
# generated theora_image_transportConfig.cmake
set(libraries "theora_image_transport;/opt/ros_android/target/lib/libopencv_stitching3.a;/opt/ros_android/target/lib/libopencv_superres3.a;/opt/ros_android/target/lib/libopencv_videostab3.a;/opt/ros_android/target/lib/libopencv_aruco3.a;/opt/ros_android/target/lib/libopencv_bgsegm3.a;/opt/ros_android/target/lib/libopencv_bioinspired3.a;/opt/ros_android/target/lib/libopencv_ccalib3.a;/opt/ros_android/target/lib/libopencv_dpm3.a;/opt/ros_android/target/lib/libopencv_face3.a;/opt/ros_android/target/lib/libopencv_photo3.a;/opt/ros_android/target/lib/libopencv_fuzzy3.a;/opt/ros_android/target/lib/libopencv_img_hash3.a;/opt/ros_android/target/lib/libopencv_line_descriptor3.a;/opt/ros_android/target/lib/libopencv_optflow3.a;/opt/ros_android/target/lib/libopencv_reg3.a;/opt/ros_android/target/lib/libopencv_rgbd3.a;/opt/ros_android/target/lib/libopencv_saliency3.a;/opt/ros_android/target/lib/libopencv_stereo3.a;/opt/ros_android/target/lib/libopencv_structured_light3.a;/opt/ros_android/target/lib/libopencv_phase_unwrapping3.a;/opt/ros_android/target/lib/libopencv_surface_matching3.a;/opt/ros_android/target/lib/libopencv_tracking3.a;/opt/ros_android/target/lib/libopencv_plot3.a;/opt/ros_android/target/lib/libopencv_datasets3.a;/opt/ros_android/target/lib/libopencv_text3.a;/opt/ros_android/target/lib/libopencv_dnn3.a;/opt/ros_android/target/share/OpenCV-3.3.1-dev/3rdparty/lib/liblibprotobuf.a;/opt/ros_android/target/lib/libopencv_xfeatures2d3.a;/opt/ros_android/target/lib/libopencv_ml3.a;/opt/ros_android/target/lib/libopencv_shape3.a;/opt/ros_android/target/lib/libopencv_video3.a;/opt/ros_android/target/lib/libopencv_ximgproc3.a;/opt/ros_android/target/lib/libopencv_calib3d3.a;/opt/ros_android/target/lib/libopencv_features2d3.a;/opt/ros_android/target/lib/libopencv_flann3.a;/opt/ros_android/target/lib/libopencv_highgui3.a;/opt/ros_android/target/lib/libopencv_videoio3.a;/opt/ros_android/target/lib/libopencv_xobjdetect3.a;/opt/ros_android/target/lib/libopencv_objdetect3.a;/opt/ros_android/target/lib/libopencv_xphoto3.a;/opt/ros_android/target/lib/libopencv_imgcodecs3.a;/opt/ros_android/target/lib/libopencv_imgproc3.a;/opt/ros_android/target/lib/libopencv_core3.a;dl;m;log;/opt/ros_android/target/share/OpenCV-3.3.1-dev/3rdparty/lib/libtegra_hal.a;/opt/ros_android/target/share/OpenCV-3.3.1-dev/3rdparty/lib/liblibjpeg.a;/opt/ros_android/target/share/OpenCV-3.3.1-dev/3rdparty/lib/liblibwebp.a;/opt/ros_android/target/share/OpenCV-3.3.1-dev/3rdparty/lib/libcpufeatures.a;/opt/ros_android/target/share/OpenCV-3.3.1-dev/3rdparty/lib/liblibpng.a;/opt/ros_android/target/share/OpenCV-3.3.1-dev/3rdparty/lib/liblibtiff.a;/opt/ros_android/target/share/OpenCV-3.3.1-dev/3rdparty/lib/liblibjasper.a;/opt/ros_android/target/share/OpenCV-3.3.1-dev/3rdparty/lib/libIlmImf.a;/opt/android/sdk/android-ndk-r18b/platforms/android-24/arch-arm64/usr/lib/libz.so;theora;theoraenc;theoradec;ogg")
```

File: get_library.sh (hunk 1 above)
Comment (ivanpauno) [reply get_library.sh]: corrected

**Review by jubeira — APPROVED** [review]
Tested on my side as well, LGTM!

**Pull request merged**